### PR TITLE
Consolidate the logs section in the documentation

### LINKED ISSUE #194: Update documentation to access docker and Jupyter logs and error message
For troubleshooting, we sometimes need to access logs produced by the user Jupyter server and / or docker container.

It would be useful to update the documentation with possible strategies to access logs.

### PR DESCRIPTION
**Pull request by jtpio**
Fixes #194 

This PR consolidates the docs about the looking at logs:

image

- [x] Add / update the documentation

### CODE REVIEW

File: docs/troubleshooting/index.rst [hunk 1]
```diff
@@ -23,6 +23,64 @@ Looking at logs
...
+
+.. code-block:: bash
+
+   sudo journalctl -u jupyterhub
+   sudo journalctl -u traefik
+
+
+.. note::
+
+   To follow the logs of the services and print more content on the screen, you can combine the ``-f`` and ``-n`` options:
```
Comment (pierrepo) [inline docs/troubleshooting/index.rst:42]: Could remind the user what -f and -n options are for?
For instance:
```
To follow the logs of the services and print more content on the screen, you can combine the ``-f`` (to print the end of the logs) and ``-n`` (to specify the number of lines to display) options:
```

Comment (jtpio) [reply docs/troubleshooting/index.rst]: Sounds good, done in https://github.com/plasmabio/plasma/pull/200/commits/c62cca105662f613b005a311fa2914ffb9b306a4

**Comment by jtpio** [comment]
> For some reason, Docker container names are in the form jupyter-username- (with the minus sign at the end):

Yes this looks related to the name specified here: 

https://github.com/plasmabio/plasma/blob/d0b13b9450aa3e060691af801a3c2fe0b3c4cb00/tljh-plasma/tljh_plasma/__init__.py#L116

This was to accommodate the use of named servers, in case a user would run several servers at the same time. The simplest right now might be to just leave it as it is, to allow for named servers if some other deployment wants to enable them.

> It looks like `docker ps -a` does not show stopped user containers (only those which run repo2docker).

I think this is because the `remove` option is explicitly set to `True` here:

https://github.com/plasmabio/plasma/blob/d0b13b9450aa3e060691af801a3c2fe0b3c4cb00/tljh-plasma/tljh_plasma/__init__.py#L120

To avoid having lots of containers around when the servers are stopped.

**Comment by pierrepo** [comment]
> Yes this looks related to the name specified here:

[plasma/tljh-plasma/tljh_plasma/__init__.py](https://github.com/plasmabio/plasma/blob/d0b13b9450aa3e060691af801a3c2fe0b3c4cb00/tljh-plasma/tljh_plasma/__init__.py#L116)

> This was to accommodate the use of named servers, in case a user would run several servers at the same time. The simplest right now might be to just leave it as it is, to allow for named servers if some other deployment wants to enable them.

This is very reasonable. Do you think we could have a default value for `servername` (for instance `server`) to avoid having this lonely `-` at the end of the container name?

**Comment by pierrepo** [comment]
> I think this is because the remove option is explicitly set to True here:

[plasma/tljh-plasma/tljh_plasma/__init__.py](https://github.com/plasmabio/plasma/blob/d0b13b9450aa3e060691af801a3c2fe0b3c4cb00/tljh-plasma/tljh_plasma/__init__.py#L120)

> To avoid having lots of containers around when the servers are stopped.

Thanks for the explanation. This make sens. We should add a comment in the documentation telling the user containers are automatically cleaned after they're stopped.

But then how could we inspect user containers that stopped for unknown reason? Are we able to tell the spawner to remove user container that stopped with exit code 0 only?

**Comment by jtpio** [comment]
> But then how could we inspect user containers that stopped for unknown reason? Are we able to tell the spawner to remove user container that stopped with exit code 0 only?

After quickly browsing the code, I'm not sure `dockerspawner` is able to make that distinction: 

https://github.com/jupyterhub/dockerspawner/blob/main/dockerspawner/dockerspawner.py

Maybe there should instead be some kind of log retention in place to keep all the logs for a certain amount of time.

**Comment by pierrepo** [comment]
> Maybe there should instead be some kind of log retention in place to keep all the logs for a certain amount of time.

Yes. We usually now very quickly when a server crashed. So a retention of a couple of days should be sufficient.

**Comment by jtpio** [comment]
PR preview available at: https://plasmabio--200.org.readthedocs.build/en/200/troubleshooting/index.html#looking-at-logs

> Yes. We usually now very quickly when a server crashed. So a retention of a couple of days should be sufficient.

Let's track that in a different issue? There are different ways of doing, something simple would be to mount a volume in the user containers to it's available on the host.

**Comment by pierrepo** [comment]
> Let's track that in a different issue? 

Yes, sure.

> PR preview available at: https://plasmabio--200.org.readthedocs.build/en/200/troubleshooting/index.html#looking-at-logs

Oups, something weird with the markdown:

image

**Comment by jtpio** [comment]
Good catch, fixed in https://github.com/plasmabio/plasma/pull/200/commits/24f8389fcf083b067d5cbdc5ebed9a4a94235452

**Comment by pierrepo** [comment]
This is now OK. I let you merge.

**Pull request merged**